core/src/test/java/org/apache/calcite/sql/test/SqlOperatorBaseTest.java

```diff
@@ -7202,6 +7218,41 @@ void assertSubFunReturns(boolean binary, String s, int start,
         "nvl(CAST(NULL AS VARCHAR(6)), cast(NULL AS VARCHAR(4)))");
   }
 
+
+  private RelNode transformSqlToRel(String sql) throws Exception {
+    final SchemaPlus rootSchema = Frameworks.createRootSchema(true);
+    final SchemaPlus defSchema = rootSchema.add("hr", new HrClusteredSchema());
+    final FrameworkConfig config = Frameworks.newConfigBuilder()
+        .parserConfig(SqlParser.Config.DEFAULT)
+        .defaultSchema(defSchema)
+        .traitDefs(ConventionTraitDef.INSTANCE, RelCollationTraitDef.INSTANCE)
+        .operatorTable(SqlLibraryOperatorTableFactory.INSTANCE
+            .getOperatorTable(SqlLibrary.STANDARD, SqlLibrary.ORACLE))
+        .build();
+    Planner planner = Frameworks.getPlanner(config);
+    SqlNode parse = planner.parse(sql);
+    SqlNode validate = planner.validate(parse);
+    RelRoot planRoot = planner.rel(validate);
+    return planRoot.rel;
+
+  }
+
+  @Test public void testNvlOperands() throws Exception {
```

Comment:
The testNvlFunc has existed in SqlOperatorTest, If it doesn't have enough tests for you, please perfect it.

If this can't suit your need. You can add a SQL type test in .iq file.

Reply:
The testNvlFunc tests the function output and not the operands, same with .iq tests. Am I missing something?

Reply:
You try to !plan command in .iq file. Same as https://github.com/apache/calcite/blob/master/core/src/test/resources/sql/agg.iq#L1707 .

Reply:
Maybe in SqlToRelConverterTest. I think this test case in here is not standard.

Reply:
Thanks @NobiGo. I am new here so when I run the tests in a .iq file I get 
`No match found for function signature NVL(<CHARACTER>, <CHARACTER>)` . I guess its because NVl is supported only for Oracle Operator Tables. Is there a way to load it in the .iq tests?

Reply:
I tried to change org/apache/calcite/prepare/CalcitePrepareImpl.java:721 and added  oracle operators there, 
```
final SqlOperatorTable opTab0 =
    context.config().fun(SqlOperatorTable.class,
        SqlLibraryOperatorTableFactory.INSTANCE
            .getOperatorTable(SqlLibrary.STANDARD, SqlLibrary.ORACLE));
```
But i think we just want it configured for the Quidem tests and not in general.

Reply:
I noticed the changes in SQL to RelNode So maybe SqlToRelConverterTest(The test result in SqlToRelConverterTest.xml) is more appropriate. Of course, If you think .iq is enough for this, You can try` !use oraclefunc ` expamle as https://github.com/apache/calcite/blob/master/core/src/test/resources/sql/functions.iq#L83 .

Reply:
Thanks a lot! Yes I think .iq test is enough as we are able to verify the type of the operand using the plan